### PR TITLE
[incubator-kie-issues-1131] test migration from V7 to code generation-9

### PR DESCRIPTION
**Pull request by Abhitocode**
In addition to previous changes associated,
Migrated tests within EndEventTest.Java

The test can be identified by referring to EndEventTest.java:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java

Closes https://github.com/apache/incubator-kie-issues/issues/1131.

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3596 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3596/1/display/redirect

**Test results:**
- PASSED: 2241
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3596/1/testReport/org.kie.kogito.quarkus.workflows/AsyncAPIIT/testConsumer/">org.kie.kogito.quarkus.workflows.AsyncAPIIT.testConsumer</a></summary>
java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
</details>

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java [hunk 1]
```diff
@@ -19,58 +19,82 @@
...
-                .startProcess("ErrorEndEvent");
-        assertProcessInstanceAborted(processInstance);
-        assertThat(((org.jbpm.process.instance.ProcessInstance) processInstance).getOutcome()).isEqualTo("error");
-
+    public void testErrorEndEventProcess() {
+        Application app = ProcessTestHelper.newApplication();
+        org.kie.kogito.process.Process<ErrorEndEventModel> process = ErrorEndEventProcess.newProcess(app);
+        ProcessInstance<ErrorEndEventModel> processInstance = process.createInstance(process.createModel());
+        processInstance.start();
+        assertThat(processInstance.status()).isEqualTo(org.jbpm.process.instance.ProcessInstance.STATE_ABORTED);
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:77]: Compared to the previous version of the test, there is no check on the outcome of the process instance.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java [hunk 2]
```diff
@@ -95,93 +119,94 @@ public void produce(KogitoProcessInstance pi, String eventData) {
 
     @Test
     @Disabled("On Exit not supported, see https://issues.redhat.com/browse/KOGITO-2067")
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:121]: Test should be enabled, onEntry/onExit scripts are already supported.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java [hunk 3]
```diff
@@ -95,93 +119,94 @@ public void produce(KogitoProcessInstance pi, String eventData) {
...
+        model.setList(myList);
+        ProcessInstance<OnEntryExitScriptProcessModel> processInstance = process.createInstance(model);
+        processInstance.start();
+        assertThat(processInstance.status()).isEqualTo(org.jbpm.process.instance.ProcessInstance.STATE_COMPLETED);
         assertThat(myList).hasSize(4);
-
     }
 
     @Test
     @Disabled("On Exit not supported, see https://issues.redhat.com/browse/KOGITO-2067")
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:136]: Test should be enabled, onEntry/onExit scripts are already supported.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java [hunk 4]
```diff
@@ -95,93 +119,94 @@ public void produce(KogitoProcessInstance pi, String eventData) {
...
+        model.setList(myList);
+        ProcessInstance<OnEntryExitMixedNamespacedScriptProcessModel> processInstance = process.createInstance(model);
+        processInstance.start();
+        assertThat(processInstance.status()).isEqualTo(org.jbpm.process.instance.ProcessInstance.STATE_COMPLETED);
         assertThat(myList).hasSize(4);
-
     }
 
     @Test
     @Disabled("On Exit not supported, see https://issues.redhat.com/browse/KOGITO-2067")
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:166]: Test should be enabled, onEntry/onExit scripts are already supported.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java [hunk 5]
```diff
@@ -95,93 +119,94 @@ public void produce(KogitoProcessInstance pi, String eventData) {
...
-
-        assertProcessInstanceCompleted(processInstance);
+    public void testSignalEndWithData() {
+        Application app = ProcessTestHelper.newApplication();
+        org.kie.kogito.process.Process<EndEventSignalWithDataModel> process = EndEventSignalWithDataProcess.newProcess(app);
+        EndEventSignalWithDataModel model = process.createModel();
+        ProcessInstance<EndEventSignalWithDataModel> processInstance = process.createInstance(model);
+        processInstance.start();
 
+        assertThat(processInstance.status()).isEqualTo(org.jbpm.process.instance.ProcessInstance.STATE_COMPLETED);
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:210]: This test misses a check to verify if the end signal is actually sent and contains the data.

Comment (Abhitocode) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:210]: Will go through the test again and update

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java (hunk 1 above)
Comment (Abhitocode) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:77]: Tried the same
public void testErrorEndEventProcess() {
        Application app = ProcessTestHelper.newApplication();
        org.kie.kogito.process.Process<ErrorEndEventModel> process = ErrorEndEventProcess.newProcess(app);
        ProcessInstance<ErrorEndEventModel> processInstance = process.createInstance(process.createModel());
        processInstance.start();
        assertThat(processInstance.status()).isEqualTo(org.jbpm.process.instance.ProcessInstance.STATE_ABORTED);
        assertThat(((org.jbpm.process.instance.ProcessInstance) processInstance).getOutcome()).isEqualTo("error");
    }
    
    But resulting in error:
    [ERROR] Errors: 
[ERROR]   EndEventTest.testErrorEndEventProcess:78 ClassCast class org.jbpm.bpmn2.event.ErrorEndEventProcessInstance cannot be cast to class org.jbpm.process.instance.ProcessInstance (org.jbpm.bpmn2.event.ErrorEndEventProcessInstance and org.jbpm.process.instance.ProcessInstance are in unnamed module of loader 'app')

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java (hunk 5 above)
Comment (Abhitocode) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:210]: Do I have to implement using any signal handlers but couldnt find any signal handler inherited from kogitoWorkItemHandler, is there any other solution?

Comment (elguardian) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:205]: /jbpm-tests/src/test/java/org/jbpm/test/utils/ReceiveTaskTestHandler.java
/jbpm-tests/src/test/java/org/jbpm/test/utils/SendTaskTestHandler.java

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java (hunk 1 above)
Comment (martinweiler) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EndEventTest.java:76]: @Abhitocode the following should work:
```
assertThat(((org.kie.kogito.process.impl.AbstractProcessInstance) processInstance).internalGetProcessInstance().getOutcome()).isEqualTo("error");
```

**Review by martinweiler — APPROVED** [review]
My comment related to the `testSignalEndWithData` test still stands, but as the test did not contain such a verification to start with, I would say that's ok.

Good work @Abhitocode

**Review by elguardian — COMMENTED** [review]
LGTM

**Pull request merged**